src/Command/Rest/DisableCommand.php

```diff
@@ -96,23 +95,38 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $rest_resources_ids,
             $this->translator
         );
-        $input->setArgument('resource-id', $resource_id);
-        $rest_settings = $this->getRestDrupalConfig();
-
-        unset($rest_settings[$resource_id]);
-
-        $config = $this->configFactory->getEditable('rest.settings');
-
-        $config->set('resources', $rest_settings);
-        $config->save();
-
-        $io->success(
-            sprintf(
-                $this->trans('commands.rest.disable.messages.success'),
-                $resource_id
-            )
-        );
+        $resources = \Drupal::service('entity_type.manager')
+            ->getStorage('rest_resource_config')->loadMultiple();
+        if ($resources[$this->getResourceKey($resource_id)]) {
+            $routeBuilder = \Drupal::service('router.builder');
+            $resources[$this->getResourceKey($resource_id)]->delete();
+            // Rebuild routing cache.
+            $routeBuilder->rebuild();
+
+            $io->success(
+                sprintf(
+                    $this->trans('commands.rest.disable.messages.success'),
+                    $resource_id
+                )
+            );
+            return true;
+        }
+        $message = sprintf($this->trans('commands.rest.disable.messages.already-disabled'), $resource_id);
```

Comment:
This is a new string to translate, I am adding this because if you try to disable a rest which is already disabled then a message like 
Rest ID "%s" was already disabled

I think makes sense, 
else
since it is a new string to translate(and takes effort to do it in != languages) then I think I can remove that line and move lines 106-111 into 114 (which is out of "if" conditional) so that it displays always success message without taking care if it is already disabled
What do you think @enzolutions  ?